File: frontend/src/api/xui.ts
import api from './index'
import type { ApiResponse, PaginationParams } from '@/types'

// XUI 服务器接口
export interface XuiServer {
  id: string
  name: string
  host: string
  domain?: string
  port: number
  username: string
  password?: string
  is_ssl: boolean
  web_path: string
  status: number
  cert_file?: string
  key_file?: string
  remark?: string
  create_time: string
  update_time: string
}

// XUI 入站接口
export interface XuiInbound {
  id: string
  server_id: string
  inbound_id: number
  listen_host: string
  listen_port: number
  protocol: number
  remark?: string
  status: number
  default_username: string
  default_password?: string
  create_time: string
  update_time: string
}

// XUI 账号接口
export interface XuiAccount {
  id: string
  email: string
  uuid: string
  enable: boolean
  flow: string
  limit_ip: number
  total_gb: number
  expire_time: number
  up: number
  down: number
  inbound_ids: number[]
}

// ==================== XUI 服务器 API ====================

export const getXuiServerList = (params?: PaginationParams & { 
  name?: string
  host?: string
  domain?: string
  status?: number
  create_time_start?: string
  create_time_end?: string
}) => {
  return api.get<any, ApiResponse<XuiServer>>('/v1/xui/server', { params })
}

export const getXuiServerDetail = (id: string) => {
  return api.get<any, XuiServer>(`/v1/xui/server/${id}`)
}

export const createXuiServer = (data: Partial<XuiServer>) => {
  return api.post<any, XuiServer>('/v1/xui/server', data)
}

export const updateXuiServer = (id: string, data: Partial<XuiServer>) => {
  return api.put<any, XuiServer>(`/v1/xui/server/${id}`, data)
}

export const deleteXuiServer = (id: string) => {
  return api.delete(`/v1/xui/server/${id}`)
}

// ==================== XUI 入站 API ====================

export const getXuiInboundList = (params?: PaginationParams & { 
  server_id?: string
  listen_port?: number
  protocol?: number
  status?: number
  create_time_start?: string
  create_time_end?: string
}) => {
  return api.get<any, ApiResponse<XuiInbound>>('/v1/xui/inbound', { params })
}

export const getXuiInboundDetail = (id: string) => {
  return api.get<any, XuiInbound>(`/v1/xui/inbound/${id}`)
}

export const createXuiInbound = (data: Partial<XuiInbound>) => {
  return api.post<any, XuiInbound>('/v1/xui/inbound', data)
}

export const updateXuiInbound = (id: string, data: Partial<XuiInbound>) => {
  return api.put<any, XuiInbound>(`/v1/xui/inbound/${id}`, data)
}

export const deleteXuiInbound = (id: string) => {
  return api.delete(`/v1/xui/inbound/${id}`)
}

// ==================== XUI 账号管理 API ====================

export const getXuiAccountsByInbound = (inboundId: string) => {
  return api.get<any, { items: XuiAccount[] }>(`/v1/xui/account/inbound/${inboundId}`)
}

export const addAccountToInbound = (inboundId: string, accountId: string) => {
  return api.post(`/v1/xui/account/inbound/${inboundId}/account/${accountId}`)
}

export const removeAccountFromInbound = (inboundId: string, accountId: string) => {
  return api.delete(`/v1/xui/account/inbound/${inboundId}/account/${accountId}`)
}

// ==================== XUI 操作 API ====================

// 同步入站配置
export const syncXuiInbounds = (serverId: string) => {
  return api.post<any, {
    code: number
    message: string
    data: {
      server_id: string
      server_name: string
      inbound_count: number
      server_info_count: number
      created: number
      updated: number
      skipped: number
    }
  }>(`/v1/xui/operation/sync-inbounds/${serverId}`)
}

// 测试服务器连接
export const testXuiConnection = (serverId: string) => {
  return api.post<any, {
    code: number
    message: string
    data: {
      success: boolean
      message: string
    }
  }>(`/v1/xui/operation/test-connection/${serverId}`)
}


// ==================== XUI 操作日志 API ====================

export interface XuiOperationLog {
  id: string
  inbound_id: string
  inbound_info: string
  account_id: string
  account_username: string
  error_message: string
  retry_count: number
  is_resolved: boolean
  create_time: string
}

// 获取失败的操作日志
export const getFailedLogs = (params?: PaginationParams & {
  inbound_id?: string
}) => {
  return api.get<any, ApiResponse<XuiOperationLog>>('/v1/xui/account/failed-logs', { params })
}

// 重试失败的操作
export const retryFailedLog = (logId: string) => {
  return api.post<any, {
    code: number
    message: string
    data: {
      success: boolean
      message: string
      result?: any
    }
  }>(`/v1/xui/account/retry-failed/${logId}`)
}

// 批量重试失败的操作
export const batchRetryFailedLogs = (inboundId?: string) => {
  return api.post<any, {
    code: number
    message: string
    data: {
      total: number
      success: number
      failed: number
    }
  }>('/v1/xui/account/batch-retry-failed', null, {
    params: inboundId ? { inbound_id: inboundId } : undefined
  })
}

// 批量添加账号到入站
export const batchAddAccountsToInbound = (inboundId: string, accountIds: string[]) => {
  return api.post<any, {
    code: number
    message: string
    data: {
      total: number
      success: number
      failed: number
      success_list: Array<{ account_id: string; username: string }>
      failed_list: Array<{ account_id: string; username: string; error: string }>
    }
  }>('/v1/xui/account/batch-add', {
    inbound_id: inboundId,
    account_ids: accountIds
  })
}

// 将账号添加到所有入站
export const addAccountToAllInbounds = (accountId: string) => {
  return api.post<any, {
    code: number
    message: string
    data: {
      total: number
      success: number
      failed: number
      failed_list: Array<{ inbound_id: string; inbound_info: string; error: string }>
      is_all_added: boolean
    }
  }>(`/v1/xui/account/add-to-all-inbounds/${accountId}`)
}

// 从所有入站删除账号
export const removeAccountFromAllInbounds = (accountId: string) => {
  return api.delete<any, {
    code: number
    message: string
    data: {
      total: number
      success: number
      failed: number
      failed_list: Array<{ inbound_id: string; inbound_info: string; error: string }>
      is_all_removed: boolean
    }
  }>(`/v1/xui/account/remove-from-all-inbounds/${accountId}`)
}

File: backend/app/models/xui.py
"""
XUI 面板相关数据模型
"""
from enum import IntEnum
from tortoise import fields
from .base import BaseModel


class XuiStatus(IntEnum):
    """XUI 状态"""
    ACTIVE = 1  # 正常
    INACTIVE = 2  # 停用
    ERROR = 3  # 异常


class XuiProtocol(IntEnum):
    """代理协议"""
    HTTP = 1  # HTTP
    SOCKS = 2  # SOCKS5


class XuiServer(BaseModel):
    """
    XUI 服务器配置
    """
    name = fields.CharField(max_length=50, description='服务器名称')
    host = fields.CharField(max_length=50, index=True, description='服务器地址（IP）')
    domain = fields.CharField(max_length=100, index=True, null=True, description='域名（用于 HTTPS 访问）')
    port = fields.IntField(default=10010, description='XUI 面板端口')
    username = fields.CharField(max_length=50, description='XUI 登录用户名')
    password = fields.TextField(description='XUI 登录密码（加密存储）')
    is_ssl = fields.BooleanField(default=False, description='是否使用 HTTPS')
    web_path = fields.CharField(max_length=50, default='/web3', description='Web 路径前缀')
    status = fields.IntEnumField(XuiStatus, default=XuiStatus.ACTIVE, description='状态')
    
    # SSL 证书配置
    cert_file = fields.CharField(max_length=255, null=True, description='SSL 证书文件路径')
    key_file = fields.CharField(max_length=255, null=True, description='SSL 私钥文件路径')
    
    # 备注
    remark = fields.TextField(null=True, description='备注')

    class Meta:
        table = "xui_server"
        table_description = "XUI 服务器配置"
        ordering = ["-create_time"]
        indexes = [
            ("status", "create_time"),
            ("host",),
            ("domain",),
        ]

    def __repr__(self):
        return f"<XuiServer(id={self.id}, name={self.name}, host={self.host})>"

    __str__ = __repr__


class XuiInbound(BaseModel):
    """
    XUI 入站配置
    """
    server = fields.ForeignKeyField(
        "models.XuiServer",
        related_name="inbounds",
        description='关联的 XUI 服务器'
    )
    
    inbound_id = fields.IntField(description='XUI 面板中的入站 ID')
    listen_host = fields.CharField(max_length=50, description='监听地址')
    listen_port = fields.IntField(index=True, description='监听端口')
    protocol = fields.IntEnumField(XuiProtocol, description='协议类型')
    remark = fields.CharField(max_length=100, null=True, description='备注')
    status = fields.IntEnumField(XuiStatus, default=XuiStatus.ACTIVE, description='状态')
    
    # 认证信息（默认账号：cqrxy / Zpaily88）
    default_username = fields.CharField(max_length=50, default='cqrxy', description='默认用户名')
    default_password = fields.TextField(null=True, description='默认密码（加密存储）')
    
    # 多对多关系：入站可以有多个服务器账号
    accounts = fields.ManyToManyField(
        "models.ServerAccount",
        related_name="xui_inbounds",
        through="xuiinbound_accounts",
        description='关联的服务器账号'
    )

    class Meta:
        table = "xui_inbound"
        table_description = "XUI 入站配置"
        ordering = ["-create_time"]
        indexes = [
            ("server_id", "status"),
            ("listen_port",),
            ("server_id", "listen_host", "listen_port"),
        ]
        unique_together = [("server_id", "listen_host", "listen_port")]

    def __repr__(self):
        return f"<XuiInbound(id={self.id}, host={self.listen_host}, port={self.listen_port})>"

    __str__ = __repr__
